Frameworks/Foundation/NSFileManager.mm

```diff
@@ -142,11 +151,32 @@ - (NSArray*)URLsForDirectory:(NSSearchPathDirectory)directory inDomains:(NSSearc
 // Locating Application Group Container Directories
 
 /**
- @Status Stub
+ @Status Caveat
+ @Notes Container is contained within a Publisher Cache Folder, so a user could erase the data
 */
 - (NSURL*)containerURLForSecurityApplicationGroupIdentifier:(NSString*)groupIdentifier {
-    UNIMPLEMENTED();
-    return nil;
+    ComPtr<IApplicationDataStatics> applicationDataStatics;
+    RETURN_NULL_IF_FAILED(Windows::Foundation::GetActivationFactory(
+        Microsoft::WRL::Wrappers::HStringReference(RuntimeClass_Windows_Storage_ApplicationData).Get(), &applicationDataStatics));
+
+    ComPtr<IApplicationData> applicationData;
+    RETURN_NULL_IF_FAILED(applicationDataStatics->get_Current(&applicationData));
+
+    // IApplicationData3 has the method needed for container URL
+    ComPtr<IApplicationData3> applicationData3;
+    RETURN_NULL_IF_FAILED(applicationData.As(&applicationData3));
+
+    ComPtr<IStorageFolder> folder;
+    RETURN_NULL_IF_FAILED(applicationData3->GetPublisherCacheFolder(Strings::NarrowToWide<HSTRING>(groupIdentifier).Get(), &folder));
+
+    // Convert to IStorageItem to get the path
+    ComPtr<IStorageItem> storageItem;
+    RETURN_NULL_IF_FAILED(folder.As(&storageItem));
+
+    HSTRING folderPath;
```

Comment:
You actually want a `Wrappers::HString` [https://msdn.microsoft.com/en-us/library/hh771141.aspx](https://msdn.microsoft.com/en-us/library/hh771141.aspx)

`HStringReference` is for a reference to an existing string resource to avoid copies.




---
In reply to: [109047974](https://github.com/Microsoft/WinObjC/pull/2384#discussion_r109047974) [](ancestors = 109047974)